src/vs/platform/workspace/browser/editSessionsStorageService.ts

```diff
@@ -0,0 +1,37 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IDisposable } from 'vs/base/common/lifecycle';
+import { Registry } from 'vs/platform/registry/common/platform';
+import { IWorkspaceFolder } from 'vs/platform/workspace/common/workspace';
+
+export interface IEditSessionContribution {
+	store(workspaceFolder: IWorkspaceFolder, result: { push(value: unknown): void }): void;
```

Comment:
passing in a delegate feels a bit weird, maybe `IEditSessionContribution<T> { store(workspaceFolder): T }` is easier to understand?